Sources/ClientRuntime/Networking/Http/URLSession/URLSessionTLSOptions.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public struct URLSessionTLSOptions {
+
+    /// Filename of the trust store certificate file in main bundle (.cer).
+    public var certificateFile: String?
+
+    /// Name of key store file (.p12).
+    public var keyStoreName: String?
+
+    /// Password for the key store if required.
+    public var keyStorePassword: String?
+
+    /// Information is provided to use custom trust store
+    public var useSelfSignedCertificate: Bool
+
+    /// Information is provided to use custom key store
+    public var useProvidedKeystore: Bool
```

Comment:
These are derived variables, but someone can modify them after the init function. Is that intended?

Reply:
Will change so they cant modify